packages/xo-remote-parser/src/index.js

```diff
@@ -2,68 +2,83 @@ import filter from 'lodash/filter'
 import trim from 'lodash/trim'
 import trimStart from 'lodash/trimStart'
-import urlParser from 'url-parse'
+import queryString from 'querystring'
+import { URL } from 'url'
 
-const NFS_RE = /^([^:]+):(?:(\d+):)?([^:]+)$/
-const SMB_RE = /^([^:]+):(.+)@([^@]+)\\\\([^\0]+)(?:\0(.*))?$/
+const NFS_RE = /^([^:]+):(?:(\d+):)?([^:?]+)(\?[^?]*)?$/
+const SMB_RE = /^([^:]+):(.+)@([^@]+)\\\\([^\0?]+)(?:\0([^?]*))?(\?[^?]*)?$/
 
 const sanitizePath = (...paths) => filter(map(paths, s => s && filter(map(s.split('/'), trim)).join('/'))).join('/')
 
-export const parse = string => {
-  const object = {}
+const parseOptionList = (optionList = '') => {
+  if (optionList?.startsWith('?')) {
+    optionList = optionList.substring(1)
+  }
+  const parsed = queryString.parse(optionList)
+  Object.keys(parsed).forEach(key => {
+    const val = parsed[key]
+    parsed[key] = val === 'true' ? true : val === 'false' ? false : JSON.parse(val)
+  })
+  return parsed
+}
+
+const makeOptionList = options => {
+  const encoded = {}
 
-  const [type, rest] = string.split('://')
+  Object.keys(options).forEach(key => {
+    const val = options[key]
+    encoded[key] = val === 'true' || val === 'false' ? val : JSON.stringify(val)
+  })
+  return queryString.stringify(encoded)
+}
+
+export const parse = string => {
+  let object = {}
+  let [type, rest] = string.split('://')
   if (type === 'file') {
     object.type = 'file'
+    let optionList
+    ;[rest, optionList] = rest.split('?')
     object.path = `/${trimStart(rest, '/')}` // the leading slash has been forgotten on client side first implementation
+    object = { ...parseOptionList(optionList), ...object }
   } else if (type === 'nfs') {
     object.type = 'nfs'
-    let host, port, path
+    let host, port, path, optionList
     // Some users have a remote with a colon in the URL, which breaks the parsing since this commit: https://github.com/vatesfr/xen-orchestra/commit/fb1bf6a1e748b457f2d2b89ba02fa104554c03df
     try {
-      ;[, host, port, path] = NFS_RE.exec(rest)
+      ;[, host, port, path, optionList] = NFS_RE.exec(rest)
     } catch (err) {
       ;[host, path] = rest.split(':')
       object.invalidUrl = true
     }
     object.host = host
     object.port = port
     object.path = `/${trimStart(path, '/')}` // takes care of a missing leading slash coming from previous version format
+    object = { ...parseOptionList(optionList), ...object }
   } else if (type === 'smb') {
     object.type = 'smb'
-    const [, username, password, domain, host, path = ''] = SMB_RE.exec(rest)
+    const [, username, password, domain, host, path = '', optionList] = SMB_RE.exec(rest)
     object.host = host
     object.path = path
     object.domain = domain
     object.username = username
     object.password = password
+    object = { ...parseOptionList(optionList), ...object }
   } else if (type === 's3' || type === 's3+http') {
-    const parsed = urlParser(string, true)
+    const parsed = new URL(string)
```

Comment:
Why did you change this?

Reply:
because this dependency was not needed anymore , and the standard `URL` do the job (since we have full urls here)

Reply:
Still, it's unrelated to this PR and it was compatible with browsers.

But I checked and the `url` polyfill used by Browserify when compiling for browsers is not much bigger, you can keep this change.